src/App.js

```diff
@@ -14,11 +15,100 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    generalArray: goodsFromServer,
```

Comment:
Немає бути в назві поля слів array, object, string
```suggestion
    goods: goodsFromServer,
```

Reply:
Можеш не зберігати цей масив в стейті, а для reset брати копію `goodsFromServer`